### ISSUE TITLE
Error while setting up greenchoice platform for sensor

### ISSUE THREAD

**Issue by mvgalen** [barisdemirdelen/homeassistant-greenchoice#18]
Previously the integration worked correctly. I did not change anything in the configuration. I confirmed I am able to login to the greenchoice website with the username /password as stated in the secrets.yaml.

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 361, in _async_setup_platform
    await asyncio.shield(awaitable)
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/greenchoice/sensor.py", line 99, in setup_platform
    throttled_api_update(greenchoice_api)
  File "/usr/src/homeassistant/homeassistant/util/__init__.py", line 184, in wrapper
    result = method(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/greenchoice/sensor.py", line 116, in throttled_api_update
    api_result = api.update()
                 ^^^^^^^^^^^^
  File "/config/custom_components/greenchoice/api.py", line 92, in update
    self.update_usage_values(self.result)
  File "/config/custom_components/greenchoice/api.py", line 121, in update_usage_values
    meter_readings = MeterReadings.from_dict(meter_json)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/greenchoice/model.py", line 297, in from_dict
    productTypes=[MeterProduct.from_dict(r) for r in data.get("productTypes")],
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not iterable

And another log entry:

Logger: custom_components.greenchoice.api
Source: custom_components/greenchoice/api.py:75
integration: greenchoice ([documentation](https://github.com/barisdemirdelen/homeassistant-greenchoice/), [issues](https://github.com/barisdemirdelen/homeassistant-greenchoice/issues))
First occurred: 12:11:39 (1 occurrences)
Last logged: 12:11:39

Error retrieving response!

**Comment by eythian** [comment]
After upgrading HA to the latest version, I'm seeing possibly related issues, where previously it worked fine.

```
Logger: homeassistant.components.sensor
Source: helpers/entity_platform.py:361
integration: Sensor (documentation, issues)
First occurred: October 4, 2024 at 9:57:46 PM (1 occurrences)
Last logged: October 4, 2024 at 9:57:46 PM

Error while setting up greenchoice platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 361, in _async_setup_platform
    await asyncio.shield(awaitable)
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/greenchoice/sensor.py", line 104, in setup_platform
    throttled_api_update(greenchoice_api)
  File "/usr/src/homeassistant/homeassistant/util/__init__.py", line 184, in wrapper
    result = method(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/greenchoice/sensor.py", line 121, in throttled_api_update
    api_result = api.update()
                 ^^^^^^^^^^^^
  File "/config/custom_components/greenchoice/api.py", line 200, in update
    self.update_usage_values(self.result)
  File "/config/custom_components/greenchoice/api.py", line 216, in update_usage_values
    meter_readings = self.get_meter_readings()
                     ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/greenchoice/api.py", line 109, in get_meter_readings
    meter_json = self._validate_response(
                 ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/greenchoice/api.py", line 83, in _validate_response
    raise ApiError("Error retrieving response!")
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/greenchoice/api.py", line 28, in __init__
    super(message)
TypeError: super() argument 1 must be a type, not str
```

and

```
This error originated from a custom integration.

Logger: custom_components.greenchoice.api
Source: custom_components/greenchoice/api.py:27
integration: greenchoice (documentation, issues)
First occurred: October 4, 2024 at 9:57:46 PM (1 occurrences)
Last logged: October 4, 2024 at 9:57:46 PM

Error retrieving response!
```

**Comment by barisdemirdelen** [comment]
It seems greenchoice changed its api, it doesnt work for me now as well. Will try to push a fix

**Comment by barisdemirdelen** [comment]
Should be fixed with #19
